File: C/DS/array/linear_search.c
#include "Array.h"
#include <stdio.h>

//Linear search are for 1D unsorted array to find out whether a given element is present in array or not.
//Time Complexity : O(n)
//Space Complexity : O(1)

int linear_search(int *arr, int size, int target) {

    int index = -1;

    for(int i = 0; i < size; i++) {
        if (arr[i] == target) {
            index = i;
            break;
        }
    }
    if (index == -1) 
        printf("%d is not present in  given array.\n", target);
    else
     printf("%d is present in given array at index: %d.\n", target,  index);

    return index;
}
            

File: Java/7. Searching Techniques/LinearSearch.java


public class LinearSearch {
    
    public int search(int[] arr, int target){
        if (arr.length == 0) return -1;
        for (int i = 0 ; i < arr.length; i++){
            if (arr[i] == target) return i;
        }
        return -1;
    }
}
